add_node.rb

```diff
@@ -25,24 +55,43 @@
 	puts "Enter department: "
 	department = gets.chomp
 
-	puts "Enter semester (mid spring 2016): "
-	semester = gets.chomp
+	if batch_semester.empty?
```

Comment:
Here, we **must** show the default to the user. So, move the default calculation to the top of the script and show the default here. (Think Linux command defaults.)